### PR TITLE
[WFCORE-4412] Eliminating CapabilityServiceTarget.addCapability() deprecated method usages

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFCORE-4412

### CODE REVIEW

**Comment by ropalka** [comment]
The commit was excluded as you suggested @darranl

**Comment by ropalka** [comment]
Retest this please

**Comment by ropalka** [comment]
Retest this please

**Comment by darranl** [comment]
@ropalka please don't use the test phrase unless you see a lot of errors as you kick off loads of jobs on CI that already passed.  Instead click on the details link and you can click run at the top right of the results screen to kick off the specific job again.

**Comment by ropalka** [comment]
Thanks for the hint @darranl - will do it since now on.

File: elytron/src/main/java/org/wildfly/extension/elytron/TrivialAddHandler.java [hunk 1]
```diff
@@ -58,12 +58,14 @@
         this.initialMode = checkNotNullParam("initialMode", initialMode);
```
Comment (jmesnil) [inline elytron/src/main/java/org/wildfly/extension/elytron/TrivialAddHandler.java]: @darranl @ropalka Is it ok to keep this elytron change or should it be handled by another Elytron PR?

Comment (darranl) [reply elytron/src/main/java/org/wildfly/extension/elytron/TrivialAddHandler.java]: @jmesnil this current change looks Ok to me, the issue was previously @ropalka was removing an 'unused' method that coincidentally a second PR was about to start using.

**Pull request merged**